source/docs/software/basic-programming/reading-stacktraces.rst

```diff
@@ -0,0 +1,553 @@
+
+      * ``RobotInit`` was called from a number of functions from the ``frc::`` namespace (AKA, the WPILib libraries)
+
+
+      This "call stack" window represents the state of the *stack* at the time the error happened. Each line represents one method, which was *called by* the method right below it.
+
+      The examples in this page assume you are running code examples in simulation, with the debugger connected and watching for unexpected errors. Similar techniques should apply while running on a real robot.
+
+
+Perform Code Analysis
+^^^^^^^^^^^^^^^^^^^^^
+
+Once you've found the stack trace, and found the lines of code which are triggering the unhandled exception, you can start the process of determining root cause.
+
+Often, just looking in (or near) the problematic location in code will be fruitful. You may notice things you forgot, or lines which don't match an example you're referencing.
+
+.. note:: Developers who have lots of experience working with code will often have more luck looking at code than newer folks. That's ok, don't be discouraged! The experience will come with time.
+
+A key strategy for analyzing code is to ask the following questions:
+
+ * When was the last time the code "worked" (IE, didn't have this particular error)?
+ * What has changed in the code between the last working version, and now?
+
+Frequent testing and careful code changes help make this particular strategy more effective.
+
+Run the Single Step Debugger
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sometimes, just looking at code isn't enough to spot the issue. The :ref:`single-step debugger <docs/software/vscode-overview/debugging-robot-program:Debugging a Robot Program>` is a great option in this case - it allows you to inspect the series of events leading up to the unhandled exception.
+
+Search for More Information
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`Google <https://google.com>`_ is a phenomenal resource for understanding the root cause of errors. Searches involving the programming language and the name of the exception will often yield good results on more explanations for what the error means, how it comes about, and potential fixes.
+
+Seeking Outside Help
+^^^^^^^^^^^^^^^^^^^^
+
+If all else fails, you can seek out advice and help from others (both in-person and online). When working with folks who aren't familiar with your codebase, it's very important to provide the following information:
+
+ * Access to your source code, (EX: :ref:`on github.com <docs/software/basic-programming/git-getting-started:Git Version Control Introduction>`)
+ * The **full text** of the error, including the full stack trace.
+
+Common Examples & Patterns
+--------------------------
+
+There are a number of common issues which result in runtime exceptions.
+
+Null Pointers and References
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Both C++ and Java have the concept of "null" - a reference which has not yet been initialized, and does not refer to anything meaningful.
```

Comment:
In C++, references cannot be null, only pointers can.

Reply:
cranked the hand-waving up to 11 to sidestep this.